src/memory/memory.cpp

```diff
@@ -16,6 +16,12 @@ void Memory::readRaw(paddress_t addr, byte_t *dst, size_t nbytes) const {
   std::memcpy(dst, m_mem.data() + addr, nbytes);
 }
 
+word_t Memory::readWord(paddress_t addr, size_t nbytes) const {
+  word_t cur_word = 0;
+  readRaw(addr, reinterpret_cast<byte_t*>(&cur_word), nbytes);
```

Comment:
I think we can do without memcpy (since we already use reinterpret_cast =) ) relying on the fact that the word size is fixed.